### DISCUSSION TITLE
Workflow files in `processes` folder and DB

### DISCUSSION THREAD

**Discussion by MaheshkumarSundaram** [bpmnServer/bpmn-server#97] [General]
Hi @ralphhanna 

Currently, `bpmn-server` reads the BPMN files inside `processes` folder initially at the start and rewrites them to MongoDB. Also, a BPMN can be added or changed from `bpmn-client` by using `server.definitions.import`.

I would like make a change to `bpmn-server` as a customization on my end.

**Current Scenario:**

I have a separate React front-end running on a different webserver which is used to design BPMN workflows and save them in a MySQL DB, While saving to MySQL, it's also uses `server.defintions.import` of `bpmn-client` to upload the files into `bpmn-server` which writes to `processes` folder and in turn to `wf_models` in MongoDB.

![image](https://github.com/ralphhanna/bpmn-server/assets/55505611/fe9b7955-5900-4638-8dae-216f5365aab9)

**Expected Scenario:**

![image](https://github.com/ralphhanna/bpmn-server/assets/55505611/e97d09c8-8859-4a12-9efc-e26462717542)

I would like to eliminate `processes` folder from the overall picture.

**_Why?_**

Because I am using an external front-end unrelated to `bpmn-server` to design and save workflows, the `processes` folder is no longer convenient because `wf_models`  in MongoDB gets recreated everytime `bpmn-server` starts/restarts based on the contents of that folder. I don't want `wf_models` to get recreated or change unless explicitly intructed using `bpmn-client`. The permanent stores for BPMN workflows should be MySQL DB and `wf_models` in MongoDB not the `processes` folder.

But I am not sure how would I approach this! Could you please guide me on this scenario?

Thank you.

**Comment by ralphhanna** [comment]
To accomplish the above:
1. Create your own class, say 'MySQLModelsStore' you can copy it from 'ModelsDatastoreDB' class
```
class MySQLModelsDatastore implements IModelsDatastore {

```
2. In your configuration.ts file 
```
		definitions: function (server) {
			return new MySQLMOdelsDataStore(server);  // instead of ModelsDatastore(server);
		},			
```

or if you just want to avoid using processes folders and use only MongoDB (in release 1.3.25)

```
		definitions: function (server) {
			return new ModelsDatastoreDB(server);
		},			
```

**Reply by MaheshkumarSundaram** [reply]
Hi @ralphhanna 

Perfect. Thank you.
